src/Razor/src/Microsoft.NET.Sdk.Razor/ReferenceResolver.cs

```diff
@@ -46,11 +49,17 @@ public IReadOnlyList<string> ResolveAssemblies()
             return applicationParts;
         }
 
-        private DependencyClassification Resolve(ClassifiedAssemblyItem classifiedItem)
+        private void Resolve(ClassifiedAssemblyItem classifiedItem, HashSet<string> visited)
```

Comment:
Would it be an improvement to make it static? It's little strange to me that this method has some *per-task-state* that's passed as a parameter (`visited`) and some that's a member `Lookup`.

Reply:
Also, do you need the hashset anymore? Doesn't the dictionary do the job for cycle-breaking?

Reply:
We rely on this method calling the overriden method on the test class to do some work. Making it static makes it trickier 

The dictionary doesn't particularly help here - it's initialized to capture all assemblies that were passed to this type. What we want to capture is a recursive relationship when traversing dependencies.

Reply:
Is that really important though? This just seems like, surprising complicated for code that does a recursive graph walk with cycle breaking.

Reply:
I'm not sure how I could stop having two separate lookups, but I did attempt to simplify the code. Perhaps this helps.